### ISSUE TITLE
Inputs for testing the new code of plume rise with the anthropogenic point sources

### ISSUE THREAD

**Issue by JianpingHuang-NOAA** [NOAA-EMC/AQM#48]
## Description
We received a new source code from Raffaele for handling plume rise with anthropogenic point sources.  

The updated code can be found from Raffaele's github link

feature/point-source at https://github.com/rmontuoro/ufs-weather-model

We need ARL work with Chan-Hoo on updating the workflow to create the point-source input files in support of running the updated AQM code. 

In addition, we need Chan-Hoo to update the aqm.rc file as well. 

Thanks all for the support

**Comment by BrianCurtis-NOAA** [comment]
I am working on a PR for UFSWM. Right now the RT regional_atmaq runs to completion, but the regional_atmaq_debug does not.
Location of work to go into PR: https://github.com/BrianCurtis-NOAA/ufs-weather-model/tree/feature/point-source

Note for debug run in UFSWM (also @rmontuoro) :
rundir: /scratch1/NCEPDEV/stmp2/Brian.Curtis/FV3_RT/rt_129517/regional_atmaq_debug

floating point overflow:
```
244: ==== backtrace (tid: 116559) ====
244:  0 0x000000000004d455 ucs_debug_print_backtrace()  ???:0
244:  1 0x000000000396aa5d getdep_v_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM
/src/model/CMAQ/CCTM/src/aero/aero6/aero_depv.F:736
244:  2 0x000000000395c3ac aero_depv_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/model/CMAQ/CCTM/src/aero/aero6/aero_depv.F:327
244:  3 0x00000000035396ac depv_defn_mp_get_depv_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/model/CMAQ/CCTM/src/depv/m3dry/DEPV_DEFN.F:865
244:  4 0x00000000033f38f5 vdiff_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/model/CMAQ/CCTM/src/vdiff/acm2/vdiffproc.F:338
244:  5 0x0000000002f470b6 cmaq_mod_mp_cmaq_advance_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/drv/cmaq_mod.F90:160
244:  6 0x0000000002f4528f cmaq_model_mod_mp_cmaq_model_advance_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/drv/cmaq_model_mod.F90:130
244:  7 0x0000000002ed028f aqm_comp_mod_mp_aqm_comp_advance_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/aqm_comp_mod.F90:202
244:  8 0x0000000002ece62e aqm_mp_modeladvance_()  /scratch1/NCEPDEV/nems/Brian.Curtis/git/BrianCurtis-NOAA/ufs-weather-model/AQM/src/aqm_cap.F90:594
```

**Comment by JianpingHuang-NOAA** [comment]
@bbakernoaa I am testing the Raffaele's new code for the grid_spec_RRFS_CONUScompact_13km.nc on 20221212 at 06z cycle but NEXUS run failed. 

Three run log files can be found at /lfs/h2/emc/ptmp/jianping.huang/emc.para/output/20221212
 nexus_emission_2022121206_s02.id_1670952622.log
nexus_emission_2022121206_s01.id_1670952622.log
 nexus_emission_2022121206_s00.id_1670952622.log

The nexus run dirs
/lfs/h2/emc/ptmp/jianping.huang/emc.para/tmp/nexus_emission_00.32956776.dbqs01
/lfs/h2/emc/ptmp/jianping.huang/emc.para/tmp/nexus_emission_01.32956772.dbqs01
/lfs/h2/emc/ptmp/jianping.huang/emc.para/tmp/nexus_emission_02.32956775.dbqs01

Can you take a look? Thanks !

**Comment by bbakernoaa** [comment]
@JianpingHuang-NOAA Can you resubmit.  I don't see anything that shows an error really.  It just seemed to stop.

**Comment by bbakernoaa** [comment]
@JianpingHuang-NOAA Nothing is showing up in the PET logs nor the other logs.  The only thing in the error file is some MPICH error and it abruptly stopped. No indication there was an error with anything.

**Comment by JianpingHuang-NOAA** [comment]
@bbakernoaa  Now I am going back to test the large domain, G793  and NEXUS got the following error message 

/lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/NEI2016v1/v2022-07/11/NEI2016v1_0.1x0.1_????????_airports.nc
/lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/NEI2016v1/v2022-07/12/NEI2016v1_0.1x0.1_????????_airports.nc
**/lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/NEI2016v1/v2022-07/13**/NEI2016v1_0.1x0.1_????????_airports.nc
ONLY 4 DAYS
ONLY 4 DAYS
Traceback (most recent call last):
  File "./nexus_nei2016_linker.py", line 238, in <module>
    fname = get_closest_file(d, dates, files)
  File "./nexus_nei2016_linker.py", line 136, in get_closest_file
    index = iwd_in_month.index(tiwd)
ValueError: 1 is not in list

It is strange that the script is looking for Month 13. 

The detail can be found from 
/lfs/h2/emc/ptmp/jianping.huang/emc.para/output/20221212 > vim nexus_emission_2022121206_s02.id_1670955159.log

**Comment by bbakernoaa** [comment]
@JianpingHuang-NOAA that is weird but easily fixed.  I'll modify the script to check if it is December or January.  It looks from +-1 month to find the closest date and we just need to be careful in December and January.

**Comment by bbakernoaa** [comment]
@BrianCurtis-NOAA This has been fixed 👍 can close https://github.com/noaa-oar-arl/NEXUS/pull/23

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA This has been fixed +1 can close [noaa-oar-arl/NEXUS#23](https://github.com/noaa-oar-arl/NEXUS/pull/23)

10-4

**Comment by BrianCurtis-NOAA** [comment]
Yes, I see this is a placeholder for the point source code from Raffaele. Thanks @JianpingHuang-NOAA

**Comment by JianpingHuang-NOAA** [comment]
This is not fixed. We are still testing the new PT code. It will require some changes of workflow once the preliminary test is done

**Comment by bbakernoaa** [comment]
@JianpingHuang-NOAA I have already created a draft PR of the necessary workflow changes (see above).  It will require the hash numbers be updated to include the ufs-weather-model, AQM, and AQM-utils to be updated once the corresponding PRs are merged.  

Already have one PR for AQM-Utils to pull the necessary changes for the point-source preprocessor 
https://github.com/NOAA-EMC/AQM-utils/pull/3

**Comment by bbakernoaa** [comment]
> @bbakernoaa Now I am going back to test the large domain, G793 and NEXUS got the following error message
> 
> /lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/NEI2016v1/v2022-07/11/NEI2016v1_0.1x0.1_????????_airports.nc /lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/NEI2016v1/v2022-07/12/NEI2016v1_0.1x0.1_????????_airports.nc **/lfs/h2/emc/physics/noscrub/jianping.huang/data/emissions/nexus/NEI2016v1/v2022-07/13**/NEI2016v1_0.1x0.1_????????_airports.nc ONLY 4 DAYS ONLY 4 DAYS Traceback (most recent call last): File "./nexus_nei2016_linker.py", line 238, in fname = get_closest_file(d, dates, files) File "./nexus_nei2016_linker.py", line 136, in get_closest_file index = iwd_in_month.index(tiwd) ValueError: 1 is not in list
> 
> It is strange that the script is looking for Month 13.
> 
> The detail can be found from /lfs/h2/emc/ptmp/jianping.huang/emc.para/output/20221212 > vim nexus_emission_2022121206_s02.id_1670955159.log

I was specifically talking about this being solved

**Comment by JianpingHuang-NOAA** [comment]
@chan-hoo  As described at the beginning of this issue, I am testing Raffaele's PT code with your online-cmaq workflow.

I did update the fms version from 2022.01 to 2022.04 to be consistent with the UFSWM module list. However, I met the following error message. 

-- Detected ESMF Beta snapshot v8.3.0b09
-- Found ESMF: /apps/ops/para/libs/intel/19.1.3.304/cray-mpich/8.1.7/esmf/8.3.0b09/lib/libesmf.a (found suitable version "8.3.0", minimum required is "8.3.0") 
CMake Error at CMakeLists.txt:124 (find_package):
  By not providing "FindFMS.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "FMS", but
  CMake did not find one.

  Could not find a package configuration file provided by "FMS" (requested
  version 2022.04) with any of the following names:

    FMSConfig.cmake
    fms-config.cmake

  Add the installation prefix of "FMS" to CMAKE_PREFIX_PATH or set "FMS_DIR"
  to a directory containing one of the above files.  If "FMS" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
See also "/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.30/build/sorc/ufs-weather-model/src/ufs-weather-model-build/CMakeFiles/CMakeOutput.log".
make[2]: *** [sorc/CMakeFiles/ufs-weather-model.dir/build.make:91: sorc/ufs-weather-model/src/ufs-weather-model-stamp/ufs-weather-model-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:893: sorc/CMakeFiles/ufs-weather-model.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

Can you take a look? 
=======================================
Another issue, the UFSWM  (Raffaele's code, see the beginning of the issue) was compiled successfully without using the online-cmaq workflow, but got a failure when I submitted forecast model.   Here is the error message. 
+ /lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.30/ush/load_modules_run_task.sh run_fcst /lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.30/jobs/JREGIONAL_RUN_FCST
Lmod has detected the following error: The following module(s) are unknown:
"mod_ufs-weather-model"

Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore-cache load "mod_ufs-weather-model"

Also make sure that all modulefiles written in TCL start with the string
#%Module
==========================

My workflow dir is located on Cactus at 

/lfs/h2/emc/physics/noscrub/jianping.huang/nwdev/packages/aqm.v7.0.30

Thanks,

Jianping

**Comment by JianpingHuang-NOAA** [comment]
adding Brian now @BrianCurtis-NOAA

**Comment by bbakernoaa** [comment]
@chan-hoo I also am having this problem but only on WCOSS2.  Changing the module file for hera fixed the build issue there.

**Comment by chan-hoo** [comment]
It should work on wcoss2 now with the latest hash.

**Comment by JianpingHuang-NOAA** [comment]
It works, thanks @chan-hoo

**Comment by HaixiaLiu-NOAA** [comment]
@JianpingHuang-NOAA can we close this now?

**Comment by HaixiaLiu-NOAA** [comment]
According to Jianping, the current workflow does not have the capability to run with the new point source code. Once @BrianCurtis-NOAA merges the point source code changes into the UFSWM, the updates to aqm.rc and cofig.yaml files are needed. @bbakernoaa and @chan-hoo, could you work together to get this done? Thank you very much

**Comment by JianpingHuang-NOAA** [comment]
@bbakernoaa  Can you provide an updated amq.rc file to ChanHoo to update the online-cmaq workflow in support  of running the point-source code?

**Comment by bbakernoaa** [comment]
There is a draft PR already
https://github.com/ufs-community/ufs-srweather-app/pull/524 this should
have all changes already needed for point sources

Barry Baker

National Oceanic and Atmospheric Administration
Air Resources Laboratory
Physical Research Scientist
Chemical Modeling and Emissions Group Leader
NCWCP, R/ARL, Rm. 4204
5830 University Research Court
College Park, Maryland  20740
Phone: ‪(301) 683-1395‬

**Comment by HaixiaLiu-NOAA** [comment]
@chan-hoo Would you please take a look at the PR created by @bbakernoaa and work with Barry to merge the changes needed? Thank you.

**Comment by chan-hoo** [comment]
@HaixiaLiu-NOAA, this PR (#524) is not ready for review yet. Its status is 'Draft'.

**Comment by JianpingHuang-NOAA** [comment]
@chan-hoo  I checked the latest workflow and did not see the updated aqm.rc file with the point-source code.  Can you provide an update on the status of this issue?  Thanks !

**Comment by chan-hoo** [comment]
@JianpingHuang-NOAA, @bbakernoaa's PR (https://github.com/ufs-community/ufs-srweather-app/pull/524) is under review. Once it is merged, the file will be updated.

**Comment by chan-hoo** [comment]
@HaixiaLiu-NOAA @JianpingHuang-NOAA, Barry's PR has been merged and other issues have been resolved. The new point source option is available in the workflow now. If you have any problems, please let me know.

**Comment by HaixiaLiu-NOAA** [comment]
Thank you very much @chan-hoo. @JianpingHuang-NOAA please close this issue after you test it and confirm there is no problem any more. Thanks.
